pkg/agent/controller/agent.go

```diff
@@ -219,6 +291,55 @@ func (a *Controller) serviceToRemoteServiceImport(obj runtime.Object, op syncer.
 	return serviceImport, false
 }
 
+func (a *Controller) endpointToRemoteServiceImport(obj runtime.Object, op syncer.Operation) (runtime.Object, bool) {
+	ep := obj.(*corev1.Endpoints)
+	svcExport, err := a.lighthouseClient.LighthouseV2alpha1().ServiceExports(ep.Namespace).Get(ep.Name, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		// Service Export not created yet
+		return nil, false
+	} else if err != nil {
+		// some other error. Log and requeue
+		klog.Errorf("Unable to get ServiceExport for %#v: %v", ep, err)
+		return nil, true
+	}
+
+	serviceImport, err := a.getServiceImport(svcExport)
+
+	if err != nil {
+		// Requeue
+		return nil, true
```

Comment:
Perhaps we should log the error if err is `IsNotFound`.

Reply:
Not needed, it can be a race condition between ServiceImport creation and getting endpoint event. Error in `ServiceImport` creation is already logged.

Reply:
I meant log if err **isn't** `IsNotFound` 😄